lib/id3c/cli/command/etl/manifest.py

```diff
@@ -95,55 +95,71 @@ def etl_manifest(*, db: DatabaseSession):
+                mark_skipped(db, manifest_record.id)
+                continue
+
+            # Skip a record if it has a sample barcode but the barcode doesn't match an identifer
+            if sample_barcode and not sample_identifier:
+                LOG.warning(f"Skipping sample with unknown sample barcode «{sample_barcode}»")
+                mark_skipped(db, manifest_record.id)
+                continue
+
+            # Skip a record if it has a collection barcode but the barcode doesn't match an identifer
             if collection_barcode and not collection_identifier:
                 LOG.warning(f"Skipping sample with unknown collection barcode «{collection_barcode}»")
                 mark_skipped(db, manifest_record.id)
                 continue
 
-            assert not collection_identifier \
-                or collection_identifier.set_name in expected_identifier_sets["collections"], \
-                    f"Collection identifier found in set «{collection_identifier.set_name}», not {expected_identifier_sets['collections']}" # type: ignore
+            # Validate the sample identifer and assert if a record fails
+            if sample_identifier:
+                if (manifest_record.document.get("sample_type") and
+                    manifest_record.document["sample_type"] == "rdt"):
+                    assert sample_identifier.set_name in expected_identifier_sets["rdt"], \
+                        (f"Sample identifier found in set «{sample_identifier.set_name}»," +
+                        f"not {expected_identifier_sets['rdt']}")
+                else:
+                    assert sample_identifier.set_name in expected_identifier_sets["samples"], \
+                        (f"Sample identifier found in set «{sample_identifier.set_name}», " +
+                        f"not {expected_identifier_sets['samples']}")
+
+            # Skip a record if the collection identifier is from an unexpected set
+            if collection_identifier and collection_identifier.set_name not in expected_identifier_sets["collections"]:
+                LOG.warning(f"Skipping sample because collection identifier found in set «{collection_identifier.set_name}», not \
+                    {expected_identifier_sets['collections']}")
+                mark_skipped(db, manifest_record.id)
+                continue
 
-            # Sample collection date
-            collection_date = manifest_record.document.get("date")
 
             # Upsert sample cooperatively with enrollments ETL routine
             #
-            # The details document was intentionally modified by two pop()s
-            # earlier to remove barcodes that were looked up.  The
-            # rationale is that we want just one clear place in the
+            # The details document was intentionally modified by three pop()s
+            # earlier to remove barcodes that were looked up and the date.
```

Comment:
This language will need to change once you stop the date pop()

Reply:
Fixed!